File: android/src/main/kotlin/com/gdelataillade/alarm/models/NotificationSettings.kt
package com.gdelataillade.alarm.models

import android.graphics.Color
import com.gdelataillade.alarm.generated.NotificationSettingsWire
import kotlinx.serialization.Serializable

@Serializable
data class NotificationSettings(
    val title: String,
    val body: String,
    val stopButton: String? = null,
    val icon: String? = null,
    val iconColor: Int? = null,
    val keepNotificationAfterAlarmEnds: Boolean = false,
) {
    companion object {
        fun fromWire(e: NotificationSettingsWire): NotificationSettings {
            val a = e.iconColorAlpha?.toFloat()
            val r = e.iconColorRed?.toFloat()
            val g = e.iconColorGreen?.toFloat()
            val b = e.iconColorBlue?.toFloat()

            var iconColor: Int? = null
            if (a != null && r != null && g != null && b != null) {
                iconColor = Color.argb(a, r, g, b)
            }

            return NotificationSettings(
                e.title,
                e.body,
                e.stopButton,
                e.icon,
                iconColor,
                e.keepNotificationAfterAlarmEnds,
            )
        }
    }
}

File: ios/alarm/Sources/alarm/models/NotificationSettings.swift
import Foundation

struct NotificationSettings {
    var title: String
    var body: String
    var stopButton: String?
    var keepNotificationAfterAlarmEnds: Bool

    static func from(wire: NotificationSettingsWire) -> NotificationSettings {
        // NotificationSettingsWire.icon and iconColor values are ignored
        // since we can't modify the notification icon on iOS.
        return NotificationSettings(
            title: wire.title,
            body: wire.body,
            stopButton: wire.stopButton,
            keepNotificationAfterAlarmEnds: wire.keepNotificationAfterAlarmEnds
        )
    }
}
